tensorflow/compiler/xla/service/gpu/done_event_thunk.h

```diff
@@ -0,0 +1,34 @@
+#ifndef TENSORFLOW_COMPILER_XLA_SERVICE_GPU_DONE_EVENT_THUNK_H_
+#define TENSORFLOW_COMPILER_XLA_SERVICE_GPU_DONE_EVENT_THUNK_H_
+
+#include "absl/synchronization/mutex.h"
+#include "tensorflow/compiler/xla/service/gpu/thunk.h"
+
+namespace xla {
+namespace gpu {
+
+void ResetEvents(size_t num_output);
+StatusOr<std::vector<std::unique_ptr<se::Event>>> TakeDoneEvents(
+    size_t output_index, int devices);
+
+// Thunk to sync a swap(in)
+class DoneEventThunk : public Thunk {
+ public:
+  DoneEventThunk(ThunkInfo thunk_info, size_t index);
+
+  Status Initialize(const GpuExecutable& executable,
+                    se::StreamExecutor* executor) override;
+
+  Status ExecuteOnStream(const ExecuteParams& params) override;
+
+ private:
+  using SpmdDoneEvents = absl::flat_hash_map<int, std::unique_ptr<se::Event>>;
+
+  const size_t output_index;
```

Comment:
`output_index ` is not the uuid in `worker.buffer[uuid]`, it is the rank in among all outputs of this executable. Am I understanding right?

Reply:
sure. uuid works only in Alpa runtime's hierarchy but not in XLA runtime.

Reply:
OK